base/compiler/ssair/show.jl

```diff
@@ -913,8 +943,12 @@ function show_ir(io::IO, ci::CodeInfo, config::IRShowConfig=default_config(ci);
                  pop_new_node! = Returns(nothing))
     used = stmts_used(io, ci)
     cfg = compute_basic_blocks(ci.code)
+    parent = ci.parent
+    sptypes = if parent isa MethodInstance
+        Core.Compiler.sptypes_from_meth_instance(parent)
+    else Core.Compiler.EMPTY_SPTYPES end
```

Comment:
IIRC, this code does not have valid access to Core.Compiler directly as all necessary references are supposed to come from Base.IRShow (to avoid breaking revise workflows)

Reply:
Hmm... That's a problem

There are things that are quite hard to emulate here (e.g. `Core.Compiler.widenconst`). From one POV, this is trying to display the IR in a way that is more relevant to how it will be seen by codegen - but codegen has types widened, etc. already so it can work with the standard functionality from the runtime

I'm not sure this is possible to implement otherwise

Reply:
It is fine to use them, the are just supposed to get imported by Base.IRShow

Reply:
Oh that's much simpler - I misunderstood

I wonder that means for the `Core.Compiler.length` above, but I'll adjust all the new ones anyway